app/src/main/java/org/wikipedia/descriptions/DescriptionEditActivity.java

```diff
@@ -19,26 +19,26 @@
 import org.wikipedia.util.FeedbackUtil;
 import org.wikipedia.util.ShareUtil;
 
+import static org.wikipedia.Constants.INTENT_EXTRA_INVOKE_SOURCE;
+import static org.wikipedia.Constants.InvokeSource;
 import static org.wikipedia.util.DeviceUtil.hideSoftKeyboard;
 
 public class DescriptionEditActivity extends SingleFragmentActivity<DescriptionEditFragment>
         implements DescriptionEditFragment.Callback, LinkPreviewDialog.Callback {
+
     private static final String EXTRA_TITLE = "title";
     private static final String EXTRA_REVIEW_ENABLE = "review";
     private static final String EXTRA_HIGHLIGHT_TEXT = "highlightText";
     private static final String EXTRA_INVOKE_SOURCE = "source";
     private static final String EXTRA_SOURCE_LANG_DESC = "source_desc";
-    public static final int PAGE_SOURCE = 0;
-    public static final int EDIT_TASKS_TITLE_DESC_SOURCE = 1;
-    public static final int EDIT_TASKS_TRANSLATE_TITLE_DESC_SOURCE = 2;
     private ExclusiveBottomSheetPresenter bottomSheetPresenter = new ExclusiveBottomSheetPresenter();
 
-    public static Intent newIntent(@NonNull Context context, @NonNull PageTitle title, @Nullable String highlightText, boolean reviewEnabled, int invokeSource, CharSequence sourceDesc) {
+    public static Intent newIntent(@NonNull Context context, @NonNull PageTitle title, @Nullable String highlightText, boolean reviewEnabled, InvokeSource invokeSource, CharSequence sourceDesc) {
```

Comment:
Is it possible to add `@NonNull` or `@Nullable` for the parameters?

Reply:
Doen